frameworks/helloworld/src/main/java/com/mesosphere/sdk/helloworld/scheduler/ReversePhasesCustomizer.java

```diff
@@ -12,7 +13,8 @@
 public class ReversePhasesCustomizer implements PlanCustomizer {
     @Override
     public Plan updatePlan(Plan plan) {
-        if (plan.getName().equals(Constants.DEPLOY_PLAN_NAME)) {
+        // Definitely don't reverse the uninstall plan!
+        if (!SchedulerConfig.fromEnv().isUninstallEnabled() && plan.getName().equals(Constants.DEPLOY_PLAN_NAME)) {
```

Comment:
SchedulerConfig should be passed in as a constructor param rather than building one locally. Then a mock SchedulerConfig can be passed in for testing, or if someone had a weird custom SchedulerConfig for some other reason then that could be passed in as well.